player/hot_cache.go

```diff
@@ -6,41 +6,34 @@ import (
 	"github.com/karlseguin/ccache/v2"
 )
 
+const longTTL = 365 * 24 * time.Hour
+
 type HotCache struct {
 	cache *ccache.Cache
-	ttl   time.Duration
 }
 
-var hc *HotCache
-
-func Init(size int64, ttl time.Duration) *HotCache {
-	if hc == nil {
-		hc = &HotCache{cache: ccache.New(ccache.Configure().MaxSize(size)), ttl: ttl}
-	}
-	return hc
+func NewHotCache(size int64) *HotCache {
+	return &HotCache{cache: ccache.New(ccache.Configure().MaxSize(size))}
 }
 
-func (h *HotCache) Get(key string) *reflectedChunk {
-	cachedItem := h.cache.Get(key)
-	if cachedItem != nil && !cachedItem.Expired() {
-		item := cachedItem.Value().(reflectedChunk)
-		return &item
+func (h *HotCache) Get(hash string) ReadableChunk {
+	cachedItem := h.cache.Get(hash)
+	if cachedItem == nil {
+		return nil
 	}
-	return nil
+	return cachedItem.Value().(ReadableChunk)
 }
 
-func (h *HotCache) Set(key string, chunk *reflectedChunk) {
-	h.cache.Set(key, *chunk, h.ttl)
+func (h *HotCache) Set(hash string, chunk ReadableChunk) {
+	h.cache.Set(hash, chunk, longTTL)
```

Comment:
I think there was a valid reason as to why I didn't pass it as pointer. I can't recall why but can you think of anything?

Reply:
ReadableChunk used to be an interface, with *reflectedChunk implementing that interface.

Now that ReadableChunk is a byte-slice, it's passed by reference anyway.